### PR TITLE
Update the spatial awareness architecture doc.

### PR DESCRIPTION
**Pull request by wiwei**
The old spatial awareness architecture doc got fairly specific into the specific interfaces and classes and events, which is really helpful if you already have an understanding of the system (i.e. it provides a reference to things you already know).

However, if you don't know about how everything works, what you really want is a high level introduction to what it is, how it's roughly structured, and what each rough piece is responsible for.

I also called out the specific platforms that we have inbox support for.

#4852

### CODE REVIEW

File: Documentation/Architecture/SpatialAwareness.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Spatial Awareness
+
+The spatial awareness system provides support for visualizing and reasoning over mapping data of the real
+world. It provides:
+
+1. Notifications for when meshes have been generated (and a way to get those meshes).
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
1. Notifications for when observations (ex: meshes) have been generated (and a way to get those observations).
```

File: Documentation/Architecture/SpatialAwareness.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+2. A mechanism to stop/start observation (i.e. to allow for runtime disabling of mesh visualization and generation).
+3. Mesh visualization.
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
3. Visualization of the observed environment.
```

File: Documentation/Architecture/SpatialAwareness.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This system only works on platforms that support spatial mapping. Currently MRTK provides inbox support
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
This system only works on platforms that support spatial mapping. Currently MRTK provides in-box support
```

File: Documentation/Architecture/SpatialAwareness.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+for spatial mapping on the following platforms:
+
+- HoloLens 1
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
- HoloLens
```

File: Documentation/Architecture/SpatialAwareness.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+## Breakdown
+
+The spatial awareness system is composed of multiple parts, which together work to read in and then
+expose and render underlying spatial data.
+
+### System
+
+Similar to the [input system](InputSystem/Terminology.md), at the root of the spatial awareness system
+is an object that is responsible for initializing and running the spatial work. It does so by creating
+and then running its list of spatial observers (which are also referred to as data providers in).
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
and then running its list of spatial observers (which are also referred to as data providers).
```

File: Documentation/Architecture/SpatialAwareness.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+### Observers
+
+A spatial observer is responsible for interfacing with a specific underlying system (for example, the
+inbox MRTK spatial observer talks with the Windows Mixed Reality platform to get HoloLens meshes) and
```
Comment (davidkline-ms) [inline Documentation/Architecture/SpatialAwareness.md]:
```suggestion
in-box MRTK spatial observer talks with the Windows Mixed Reality platform to get HoloLens meshes) and
```

File: Documentation/MixedRealityConfigurationGuide.md [hunk 7]
```diff
@@ -132,23 +132,19 @@ The Mixed Reality Project provides a full featured Teleportation system for mana
...
-You can view the architecture behind the [MRTK Spatial awareness system here](../Documentation/Architecture/SpatialAwareness/SpatialAwarenessSystemArchitecture.md).
-
-> The MRTK also allows you to write your own spatial awareness system and you can use the selection below to switch the system used without rewriting the toolkit. For more information on writing your own systems, [please see this guide to building registered services]() (Coming soon)
+You can view the architecture behind the [MRTK Spatial awareness system here](../Documentation/Architecture/SpatialAwareness.md).
 
 <img src="../Documentation/Images/MixedRealityToolkitConfigurationProfileScreens/MRTK_SpatialAwarenessSystemSelection.png" width="650px" style="display:block;">
 
-The Mixed Reality Toolkit spatial awareness configuration lets you tailor how the system starts, whether it is automatically when the application starts or later programmatically as well as setting the extents for the field of view.
+The Mixed Reality Toolkit spatial awareness configuration lets you tailor how the system starts, whether it is automatically when the application starts
```
Comment (davidkline-ms) [inline Documentation/MixedRealityConfigurationGuide.md]: It is the observers that have the configuration settings re: start behavior, etc. The system only manages observers.

Comment (wiwei) [reply Documentation/MixedRealityConfigurationGuide.md]: I don't think this section goes into any degree of specificity about observers and stuff - this is purely saying "hey look at this profile section in the inspector to configure spatial awareness"

**Pull request merged**